doc/src/bond_cpp.md

```diff
@@ -185,6 +185,59 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Type evolution
+==============
+
+Bond does not use explicit versioning to deal with changes to schemas over time
+Instead, Bond supports certain type evolution operations which allow the
+producer and consumer of Bond types to evolve without lockstep coordination.
+
+The following changes to a type will never break compatibility across the wire:
+
+- Adding or removing an `optional` or `optional_required` field
+- Changing type between `int32` and `enum`
+- Changing type between `vector<T>` and `list<T>`
+- Changing type between `blob` and `vector<uint8>` or `blob` and `list<uint8>`
+- Changing type between `T` and `bonded<T>`
+- Adding new enumeration values at the end of an enumeration
+
+The following changes to a type are generally safe but require some
+consideration about how the change is rolled out:
+
+- Changing between `optional` and `required_optional` or between
+`required_optional` and `required`. This facilitates a two-step process for
+changing between `optional` and `required` but the first step must be completed
```

Comment:
That seems very odd stylistically to me -- comma before the 'but' perhaps but not after. See point 2 of http://grammar.ccc.commnet.edu/Grammar/commas.htm